userspace/libsinsp/socket_handler.h

```diff
@@ -1519,14 +1523,11 @@ class socket_data_handler
 						 (pending_reqs > 10) ? sinsp_logger::SEV_WARNING : sinsp_logger::SEV_TRACE);
 		}
 
-		if(dns_cleanup(m_dns_reqs))
+		if(!dns_req_done(m_dns_reqs.get()))
 		{
-			m_dns_reqs = 0;
-		}
-		else // store for postponed canceling
-		{
-			m_pending_dns_reqs.push_back(m_dns_reqs);
+			m_pending_dns_reqs.push_back(std::move(m_dns_reqs));
 		}
+		m_dns_reqs = nullptr;
```

Comment:
Is this the actual/main fix which we weren't doing after pushing it to `m_pending_dns_reqs`?

Reply:
Yes/kinda ;)

Switching to unique_ptrs and doing std::move to push to m_pending_dns_reqs directly fixes it because the move clears m_dns_reqs. Assigning to nullptr is redundant if we went in the if block, but it's safe and the same cost as an else block.

(The push_back should be emplace_back, I'll fix that)